tests/validation/tests/v3_api/test_workload.py

```diff
@@ -858,3 +858,333 @@ def setup_project_by_role(role, remove_resource):
+            "image": TEST_IMAGE}]
+    name = random_test_name("default")
+    if role != PROJECT_READ_ONLY:
+        workload = p_client.create_workload(name=name,
+                                            containers=con,
+                                            namespaceId=ns.id,
+                                            daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        p_client.delete(workload)
+        assert len(p_client.list_workload(uuid=workload.uuid).data) == 0
+        remove_resource(workload)
+    else:
+        user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        p_client = get_project_client_for_token(project, user_token)
+        ns = rbac_get_namespace()
+        workload = p_client.create_workload(name=name,
+                                            containers=con,
+                                            namespaceId=ns.id,
+                                            daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        with pytest.raises(ApiError) as e:
+            p_client.delete(workload)
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'Forbidden'
+        remove_resource(workload)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", [CLUSTER_MEMBER,
+                                  PROJECT_MEMBER, PROJECT_OWNER,
+                                  PROJECT_READ_ONLY, CLUSTER_OWNER])
+def test_rbac_wl_daemonset_delete_negative(remove_resource):
+    if role == CLUSTER_OWNER:
+        # cluster owner can edit daemonsets in any project in the cluster
+        project = rbac_get_project()
+        user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        p2 = rbac_get_unshared_project()
+        p_client = get_project_client_for_token(project, user_token)
+        p_client2 = get_project_client_for_token(p2, user_token)
+        ns2 = rbac_get_unshared_ns()
+        name = random_test_name("default")
+        con = [{"name": "test1",
+                "image": TEST_IMAGE}]
+        workload = p_client2.create_workload(name=name,
+                                    containers=con,
+                                    namespaceId=ns.id,
+                                    daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        con = [{"name": "test1", "image": "nginx"}]
+        p_client.update(workload, containers=con)
+        validate_workload_image(p_client, workload, "nginx", ns)
+        p_client.delete(workload)
+
+    else:
+        if role == CLUSTER_MEMBER:
+            user_token = rbac_get_user_token_by_role(role)
+            project, ns = create_project_and_ns(user_token, namespace["cluster"],
+                                                random_test_name("test-rbac"))
+            p_client = get_project_client_for_token(project, user_token)
+        else:
+            project = rbac_get_project()
```

Comment:
Should this be the unshared project?
No one except the cluster owner can delete the daemon set in the unshared project.